### ISSUE TITLE
provide reason/error message for a build failure inside the current pipeline

### ISSUE THREAD

**Issue by vanous** [woodpecker-ci/woodpecker#704]
### Clear and concise description of the problem

As a last step of my pipeline (in case of failure) i would like to report the error message of the failed build without having to wrap every single step command. My reporting pipeline step would look something like this:

```yaml
  ...
  on-error:
    commands:
      - ./report_failure.sh "CI failed" CI_JOB_ERROR_MESSAGE
    when:
      status: failure
```

### Suggested solution

The `CI_JOB_ERROR_MESSAGE` or the like is what i am asking for.

### Additional context

Thank you!

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by 6543** [comment]
tldr: step fail but pipeline should pass

**Comment by vanous** [comment]
Thank you. I presume we mean the same thing... another way to tldr it:

we would like to have the **reason** for the failed status, in order to be able to send/log it to an external system
